Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM node:12-alpine as build
+ARG BUILD_DIR=rs-cart-api
+RUN mkdir -p ${BUILD_DIR}
+# Copy project files
+COPY . /${BUILD_DIR}
+WORKDIR /${BUILD_DIR}
+# Install dependencies
+RUN npm i
+RUN npm run build --prod --outputPath=./dist
```

Comment:
Source code changes more often than dependencies. Optimal solution:
1. copy package.json and package-lock.json
2. install npm dependencies
3. copy code of application
4. build application